core/src/Kokkos_View.hpp

```diff
@@ -19,13 +19,1032 @@
+           typename traits::memory_traits>;
+
+  // Compatible view of const data type
+  using const_type =
+      View<typename traits::const_data_type, typename traits::array_layout,
+           typename traits::device_type, typename traits::hooks_policy,
+           typename traits::memory_traits>;
+
+  // Compatible view of non-const data type
+  using non_const_type =
+      View<typename traits::non_const_data_type, typename traits::array_layout,
+           typename traits::device_type, typename traits::hooks_policy,
+           typename traits::memory_traits>;
+
+  // Compatible HostMirror view 
+  using host_mirror_type =
+      View<typename traits::non_const_data_type, typename traits::array_layout,
+           Device<DefaultHostExecutionSpace,
+                  typename traits::host_mirror_space::memory_space>,
+           typename traits::hooks_policy>;
+
+  // Compatible HostMirror view
+  using HostMirror = host_mirror_type;
+
+  // Unified types
+  using uniform_type = typename Impl::ViewUniformType<View, 0>::type;
+  using uniform_const_type =
+      typename Impl::ViewUniformType<View, 0>::const_type;
+  using uniform_runtime_type =
+      typename Impl::ViewUniformType<View, 0>::runtime_type;
+  using uniform_runtime_const_type =
+      typename Impl::ViewUniformType<View, 0>::runtime_const_type;
+  using uniform_nomemspace_type =
+      typename Impl::ViewUniformType<View, 0>::nomemspace_type;
+  using uniform_const_nomemspace_type =
+      typename Impl::ViewUniformType<View, 0>::const_nomemspace_type;
+  using uniform_runtime_nomemspace_type =
+      typename Impl::ViewUniformType<View, 0>::runtime_nomemspace_type;
+  using uniform_runtime_const_nomemspace_type =
+      typename Impl::ViewUniformType<View, 0>::runtime_const_nomemspace_type;
+
+  //----------------------------------------
+  // Domain rank and extents
+
+  static constexpr Impl::integral_constant<size_t, base_t::rank()> rank = {};
+  static constexpr Impl::integral_constant<size_t, base_t::rank_dynamic()>
+      rank_dynamic = {};
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+  enum {Rank KOKKOS_DEPRECATED_WITH_COMMENT("Use rank instead.") = rank()};
+#endif
+
+  KOKKOS_INLINE_FUNCTION constexpr array_layout layout() const {
+    return Impl::array_layout_from_mapping<array_layout, mdspan_type>(
+        base_t::mapping());
+  }
+
+  KOKKOS_INLINE_FUNCTION constexpr size_t stride_0() const { return stride(0); }
+  KOKKOS_INLINE_FUNCTION constexpr size_t stride_1() const { return stride(1); }
+  KOKKOS_INLINE_FUNCTION constexpr size_t stride_2() const { return stride(2); }
+  KOKKOS_INLINE_FUNCTION constexpr size_t stride_3() const { return stride(3); }
+  KOKKOS_INLINE_FUNCTION constexpr size_t stride_4() const { return stride(4); }
+  KOKKOS_INLINE_FUNCTION constexpr size_t stride_5() const { return stride(5); }
+  KOKKOS_INLINE_FUNCTION constexpr size_t stride_6() const { return stride(6); }
```

Comment:
Do these need to be `KOKKOS_INLINE_FUNCTION`, as opposed to `KOKKOS_FUNCTION`, given that they are member functions defined in the class?  The code is inconsistent on this.